### PR TITLE
My Jetpack: put My Jetpack menu item at top of Jetpack menu section

### PR DESCRIPTION
**Pull request by retrofox**
This PR puts My Jetpack menu item at top of Jetpack menu section. Also, when the site is disconnected, My Jetpack shows the connection screen allowing the user get reconnected to the site. No hard redirection.

Fixes #

#### Changes proposed in this Pull Request:

* My Jetpack: put My Jetpack menu item at top of Jetpack menu section

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Install and activate only one stand-alone plugin (e.g. Boost or Backup)
* Make sure the Jetpack menu is added and My Jetpack is the first item
* ~~Confirm that when you visit My Jetpack you are redirected to the interstitial page of the plugin you have active (e.g. `#/add-boost`)~~ Confirm that when you visit the My Jetpack page you see the regular dashboard with the product cards
* Activate a second stand-alone plugin
* Confirm that when you visit the My Jetpack page you see the regular dashboard with the product cards
* Confirm that adding products works from there
* Test differentt combinations and actions you think could be tricky

before | after
------|--------
image | image

### CODE REVIEW

**Comment by oskosk** [comment]
We'll punt this for My jetpack 1.0. I just moved it to that milestone

**Review by kraftbj — DISMISSED** [review]

**Review by kraftbj — DISMISSED** [review]
Re-approved after bringing it up to date. The original PR had a version bump that was no longer needed.

**Comment by leogermani** [comment]
Please let me have a look at this PR before merging. I want to test a few edge cases

**Review by leogermani — DISMISSED** [review]
When we disconnect Jetpack from the My Jetpack page we have to redirect the user to the first menu item under the Jetpack top level menu.

Since Jetpack is disconnected, the first menu will no longer be My Jetpack. 

Now, we end up with an error. We have to fix this and get an updated information about where to redirect the user to.

Here's where this happens: https://github.com/Automattic/jetpack/blob/master/projects/packages/my-jetpack/_inc/hooks/use-my-jetpack-connection/index.js#L15

**Comment by retrofox** [comment]
> When we disconnect Jetpack from the My Jetpack page we have to redirect the user to the first menu item under the Jetpack top level menu.

Because, by convention, the first menu item is the Jetpack plugin?

> Since Jetpack is disconnected, the first menu will no longer be My Jetpack.
> Now, we end up with an error. We have to fix this and get an updated information about where to redirect the user to.

What do you think about keeping the user in My Jetpack with a "User not connected" page?
Even more, we can use at least for now the `connection` page:

`https://testiing-site/wp-admin/admin.php?page=my-jetpack#/connection`

image

**Comment by leogermani** [comment]
> Because, by convention, the first menu item is the Jetpack plugin?

No, we redirect to whichever is the first menu item. If only Boost is active, will be Boost. If Boost and Backup are active, it will be one of the two...

> What do you think about keeping the user in My Jetpack with a "User not connected" page?

I think it makes sense, since we want to have the same experience for all plugins. This will need design, but we can start simpe and iterate

**Comment by leogermani** [comment]
Let's make the My Jetpack page visible when there's no connection and then we can merge this

**Comment by leogermani** [comment]
Ongoing discussion in p9dueE-4KC-p2

**Comment by leogermani** [comment]
Considering the discussion we had in p9dueE-4KC-p2#comment-7644 , I think we can close this PR.

**Review by retrofox — COMMENTED** [review]
testing locally, I still see the interstitial page of backup even having backup and boost installed.

image

`productsWithActivePlugin` has only one item, which is the `backup`plugin.
Something related with the local dev-env?

**Comment by leogermani** [comment]
I can't reproduce it... anything else different in your env?

**Comment by retrofox** [comment]
> [](https://github.com/Automattic/jetpack/pull/23052)I can't reproduce it... anything else different in your env?
> Uploading Captura de tela de 2022-04-04 17-07-24.png…

I can't see your image. Did you break githug? 
image

**Comment by retrofox** [comment]
> I can't reproduce it... anything else different in your env?

It doesn't work properly in my local dev-env. Probably because of how the dev-env handles the local plugins. It works was expected in a JN site

**Comment by retrofox** [comment]
actually tested again, and it works as expected. there are two Boost plugins there
image

File: projects/packages/my-jetpack/_inc/hooks/use-products/index.js
Comment (retrofox) [inline projects/packages/my-jetpack/_inc/hooks/use-products/index.js:33]: I wonder whether worths computing this `disconnectedAndOnePluginInstalled` here in the hook, because maybe it could be implemented straight to the component. Just thinking loud.

**Review by retrofox — COMMENTED** [review]
It works as expected.
However, from my PoV, showing the interstitial page of the plugin when it's the only active one doesn't make sense. It's confusing. I'd expect to see the regular My Jetpack overview page.
For this reason, I delegate the approval to @oskosk and @keoshi .

Implementation wise, I left a simple suggestion but I think it's good enough to go. Other couple of eyes won't hurt, though.

**Comment by keoshi** [comment]
First off, thanks for creating this PR! I have some thoughts and concerns about this approach, which I've openly shared before with all of your, but using this to document a few of the more tricky ones.

### Add product interstitial
When Boost (or Backup) is installed but not configured, I was not expecting to see the interstitial on the My Jetpack page. The primary action — “Add Jetpack Boost” — doesn't make sense because the plugin is already active, and it differs from the primary action on the Boost page: “Get started”.

We could do several things here, from changing the CTA, to align both views (interstitial/initial set up screen), from prioritizing the set up flow of the standalone plugin in lieu of showing My Jetpack, or even having an “empty view” for My Jetpack. My preference would be to rely on the plugin for the default flow, while showing as much of the My Jetpack page as we can.

image

image

### Jetpack plugin

After installing and setting up Jetpack, we land on Recommendations, which is part of the “Dashboard” sidebar item (or used to be), but My Jetpack is highlighted instead. Detected several issues with this:

- A click on My Jetpack is required to actually see that page.
- Clicking Dashboard on the sidebar doesn't highlight it.
- Clicking Settings afterwards highlights the Dashboard item instead.
- My Jetpack link next to Recommendations is kind of obsolete with this change, and should probably be removed.

image

**Comment by leogermani** [comment]
I've refreshed the PR and changed it so it does not redirect the user anywhere. It simply displays My Jetpack as it is even if there is only one plugin active

**Comment by jeherve** [comment]
Closing this for now because of the lack of activity on this. We can always reopen in the future if needed.

**Pull request closed without merging**
